packages/api/lambdas/sf-event-sqs-to-db-records/write-granules.js

```diff
@@ -14,43 +20,249 @@ const {
 
-const writeGranuleViaTransaction = async ({
-  cumulusMessage,
+/**
+ * Generate a Granule record to save to the core database from a Cumulus message
+ * and other contextual information
+ *
+ * @param {Object} params
+ * @param {string} params.collectionId - Collection ID for the workflow
+ * @param {Object} params.granule - Granule object from workflow message
+ * @param {Array<Object>} params.files - Granule file objects
+ * @param {number} params.collectionCumulusId
+ *   Cumulus ID of collection referenced in workflow message
+ * @param {number} params.providerCumulusId
+ *   Cumulus ID of provider referenced in workflow message
+ * @param {number} params.executionCumulusId
+ *   Cumulus ID of execution referenced in workflow message
+ * @param {number} params.pdrCumulusId
+ *   Cumulus ID of PDR referenced in workflow message
+ * @param {Object} [params.processingTimeInfo={}]
+ *   Info describing the processing time for the granule
+ * @param {Object} [params.cmrUtils=CmrUtils]
+ *   Utilities for interacting with CMR
+ * @param {number} [params.timestamp] - Timestamp for granule record. Defaults to now.
+ * @param {number} [params.updatedAt] - Updated timestamp for granule record. Defaults to now.
+ * @returns {Promise<Object>} - a granule record
+ */
+const generateGranuleRecord = async ({
+  error,
   granule,
+  files,
+  workflowStartTime,
+  workflowStatus,
   collectionCumulusId,
   providerCumulusId,
   executionCumulusId,
   pdrCumulusId,
+  processingTimeInfo = {},
+  cmrUtils = CmrUtils,
+  timestamp = Date.now(),
+  updatedAt = Date.now(),
+}) => {
+  const {
+    granuleId,
+    cmrLink,
+    published = false,
+  } = granule;
+
+  const temporalInfo = await cmrUtils.getGranuleTemporalInfo(granule);
+
+  return {
+    granule_id: granuleId,
+    status: getGranuleStatus(workflowStatus, granule),
+    cmr_link: cmrLink,
+    error,
+    published,
+    created_at: new Date(workflowStartTime),
+    updated_at: new Date(updatedAt),
+    timestamp: new Date(timestamp),
+    // Duration is also used as timeToXfer for the EMS report
+    duration: getWorkflowDuration(workflowStartTime, timestamp),
+    product_volume: getGranuleProductVolume(files),
+    time_to_process: getGranuleTimeToPreprocess(granule),
+    time_to_archive: getGranuleTimeToArchive(granule),
+    collection_cumulus_id: collectionCumulusId,
+    provider_cumulus_id: providerCumulusId,
+    execution_cumulus_id: executionCumulusId,
+    pdr_cumulus_id: pdrCumulusId,
+    // Temporal info from CMR
+    beginning_date_time: temporalInfo.beginningDateTime,
+    ending_date_time: temporalInfo.endingDateTime,
+    production_date_time: temporalInfo.productionDateTime,
+    last_update_date_time: temporalInfo.lastUpdateDateTime,
+    // Processing info from execution
+    processing_start_date_time: processingTimeInfo.processingStartDateTime,
+    processing_end_date_time: processingTimeInfo.processingEndDateTime,
+  };
+};
```

Comment:
Should you also add `data_type` and `version`?

Reply:
I left a comment on your other PR - but now I'm less sure about whether we need those

Reply:
Yeah, I rescind my earlier comment. I don't think you need to add those and we won't be adding them to the PG schema.